### PR TITLE
feat: add snapshot producer

### PR DESCRIPTION
**Pull request by pgallik**

### CODE REVIEW

File: src/BuildingRegistry.Producer.Snapshot.Oslo/ProducerBuildingProjections.cs [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    {
+        public const string TopicKey = "BuildingTopic";
+
+        private readonly IProducer _producer;
+
+        public ProducerBuildingProjections(IProducer producer, ISnapshotManager snapshotManager, string osloNamespace)
+        {
+            _producer = producer;
+
+            When<Be.Vlaanderen.Basisregisters.ProjectionHandling.SqlStreamStore.Envelope<BuildingWasMigrated>>(async (_, message, ct) =>
```
Comment (jvandaal) [inline src/BuildingRegistry.Producer.Snapshot.Oslo/ProducerBuildingProjections.cs:25]: I'm missing the handling of event `BuildingBecameUnderConstructionV2`

File: src/BuildingRegistry.Producer.Snapshot.Oslo/ProducerBuildingProjections.cs [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+                        await snapshotManager.FindMatchingSnapshot(
+                            message.Message.BuildingPersistentLocalId.ToString(),
+                            message.Message.Provenance.Timestamp,
+                            throwStaleWhenGone: false,
+                            ct),
+                    message.Position,
+                    ct);
+            });
+
+            When<Be.Vlaanderen.Basisregisters.ProjectionHandling.SqlStreamStore.Envelope<BuildingWasCorrectedFromNotRealizedToPlanned>>(async (_, message, ct) =>
```
Comment (jvandaal) [inline src/BuildingRegistry.Producer.Snapshot.Oslo/ProducerBuildingProjections.cs:280]: Why is the action body empty for the next events?

**Review by jvandaal — APPROVED** [review]

File: src/BuildingRegistry.Producer.Snapshot.Oslo/ProducerBuildingUnitProjections.cs [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+            { });
+
+            When<Be.Vlaanderen.Basisregisters.ProjectionHandling.SqlStreamStore.Envelope<BuildingWasNotRealizedV2>>(async (_, message, ct) =>
+            { });
+
+            When<Be.Vlaanderen.Basisregisters.ProjectionHandling.SqlStreamStore.Envelope<BuildingWasPlannedV2>>(async (_, message, ct) =>
+            { });
+
+            When<Be.Vlaanderen.Basisregisters.ProjectionHandling.SqlStreamStore.Envelope<BuildingWasRealizedV2>>(async (_, message, ct) =>
+            { });
```
Comment (ArneD) [inline src/BuildingRegistry.Producer.Snapshot.Oslo/ProducerBuildingUnitProjections.cs:306]: If nothing is done, remove the code?

**Pull request merged**